packages/create-amplify/src/package-manager-controller/yarn_modern_package_manager_controller.ts

```diff
@@ -0,0 +1,51 @@
+import fs from 'fs';
+import { logger } from '../logger.js';
+import { executeWithDebugLogger as _executeWithDebugLogger } from '../execute_with_logger.js';
+import { PackageManagerController } from './package_manager_controller.js';
+
+/**
+ * YarnModernPackageManagerController is an abstraction around yarn modern (yarn v2+) commands that are needed to initialize a project and install dependencies
+ */
+export class YarnModernPackageManagerController extends PackageManagerController {
+  /**
+   * constructor
+   */
+  constructor(readonly projectRoot: string) {
+    super(projectRoot);
+    this.executable = 'yarn';
+    this.binaryRunner = 'yarn';
+    this.installCommand = 'add';
+    this.initDefault = ['init', '--yes'];
+  }
+
+  initializeTsConfig = async (targetDir: string) => {
+    await this.addLockFile(targetDir);
+    await this.addTypescript(targetDir);
+    await super.initializeTsConfig(targetDir);
+  };
+
+  /**
+   * addLockFile - adds a yarn.lock file to the project root for yarn v2+
+   */
+  private addLockFile = (targetDir: string) => {
+    fs.writeFile(this.path.resolve(targetDir, 'yarn.lock'), '', (err) => {
+      if (err) {
+        logger.error(`Error creating ${targetDir}/yarn.lock ${err.message}}`);
+      } else {
+        logger.info(`${targetDir}/yarn.lock created successfully.`);
+      }
+    });
```

Comment:
I'm not sure why linter didn't catch this but we shouldn't use fs apis that use callbacks.
We await `addLockFile` above but we don't await anything inside - this creates potential race conditions.
The `fs.writeFile` (from `fs`) does not return `Promise` and can't be awaited - we should use `fs/promises` instead.
When we convert this to `fsp` we should use `try catch` here.

As for linter - mind creating GH issue to add (or fix?) linter to enforce file system api usage?

Reply:
Done a146fa1c0e

Reply:
feature request: https://github.com/aws-amplify/amplify-backend/issues/847